site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -261,10 +463,87 @@ def _metadata(self, data):
                 "rights": self._rights(data.get("license")),
                 "contributors": self._contributors(data),
                 "additional_descriptions": self._additional_descriptions(
-                    data.get("notes")
+                    {"notes": data.get("notes"), "method": data.get("method")}
+                ),
+                "locations": self._locations(data.get("locations")),
+                "subjects": self._subjects(data),
+                "version": data.get("version"),
+                "dates": self._dates(data.get("dates", [])),
+                "references": self._references(data.get("references", [])),
+                "languages": self._language(data.get("language")),
+                "related_identifiers": self._related_identifiers(
+                    data.get("related_identifiers", [])
                 ),
             }
 
+    def _access(self, data):
+        """Transform the access of a record."""
+
+        def _embargoed_record(embargo_date):
+            """Generates RDM access conditions based on legacy embargo conditions."""
+            return {
+                "record": "public",
+                "files": "restricted",
+                "embargo": {
+                    "active": True,  # boolean
+                    "until": embargo_date,  # ISO String
+                },
+            }
+
+        def _restricted_record():
+            """Generates RDM access conditions based on legacy restricted conditions."""
+            return {
+                "record": "public",
+                "files": "restricted",
+            }
+
+        def _open_record():
+            """Generates RDM access conditions for an open record."""
+            return {
+                "record": "public",
+                "files": "public",
+            }
+
+        default = _open_record()
+        access = None
+        metadata = data.get("metadata", {})
+        access_right = metadata.get("access_right")
+        embargo_date = metadata.get("embargo_date")
+
+        if not access_right:
+            # TODO throw an error actually? Or default to public?
+            return default
+
+        is_embargoed = access_right == "embargoed"
+        is_restricted = access_right == "restricted"
+        is_open = access_right == "open"
+        is_closed = access_right == "closed"
+
+        if is_open:
+            access = _open_record()
+        elif is_restricted or is_closed:
+            access = _restricted_record()
+        elif is_embargoed and embargo_date:
+            access = _embargoed_record(embargo_date)
+
+        if not access:
+            # TODO throw an error.
+            pass
+
+        return access
+
+    def _pids(self, data):
+        """Transform legacy doi in RDM pid with external provider."""
+        metadata = data.get("metadata")
+        doi = metadata.get("doi")
+
+        if not doi:
```

Comment:
I am not sure what happens here.

If the user provides a prereserved DOI, in RDM it should be seen as an external or internal pid?  In case of external, who's the provider? Datacite?

Reply:
The only difference would be the provider, right? E.g. if the passed doi is a zenodo doi, the the provider is datacite.

Reply:
Exactly, we need to translate it to the correct provider value, i.e. `datacite`. Though I think this has to happen in the service-method, since it's a separate call to the `records_service.pids` sub-service.

Reply:
You mean that the provider should be returned by the pid manager based on the identifier? E.g. [here](https://github.com/inveniosoftware/invenio-rdm-records/blob/6d7758714d08be9317bed5d402fc09731f71bf3d/invenio_rdm_records/services/pids/manager.py#L29)

Reply:
The issue is that the behaviour of passing the `pids` field as input via the `create()/update()` service methods is a bit flaky (even in the "stock" RDM API). E.g. if one passes a `doi` with `datacite` provider, and then changes over to `external`, there are additional calls necessary to un-reserve the `datacite` DOI... You can check the REST API calls in the default InvenioRDM deposit form and see that there are more calls happening to `/api/records/{id}/draft/pids`. These service calls must be "baked-in" the Zenodo legacy REST API.

Since this is actually a bit difficult to handle fully now, I would say let's make the happy paths work (i.e. a) publish with Zenodo DOI, and b) publish with an external DOI), and we tackle the edge cases separately.

Reply:
https://github.com/zenodo/zenodo-rdm/issues/353